### PR TITLE
Added Bubble Sort Doc

### PR DESCRIPTION
**Pull request by ananddasani**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

Added the Bubble sort doc, so I request @HarshCasper  to please review my PR :)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

No :)

Fixes issue #15

### CODE REVIEW

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+## Introduction
+
+Bubble sort is a simple sorting algorithm which compares two adjacent elements and swap if they are not in the correct position.
+
+## Explanation
+
+**AIM -** Repeatedly swap two adjacent elements if they are in the wrong order.
+
+**WORKING -** It works by iterating the array from the first element to the last element, comparing each pair of the elements, and swapping them if needed.
```
Comment (nidh-ish) [inline docs/sorting-algorithms/bubble-sort.mdx:14]: Make it: comparing each pair of elements, ...

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+## Algorithm
+
+- read total no. of elements say n.
+- store the elements in an array.
+- initially i = 0.
+- compare adjacent elements.
+- repeat step 4 for all n elements.
+- increment i by 1 and repeat steps 4,5 for i < n.
```
Comment (nidh-ish) [inline docs/sorting-algorithms/bubble-sort.mdx:92]: Make starting letters capital, and use number instead of no.
Keep n, i, 4, 5 in backticks.

**Review by nidh-ish — CHANGES_REQUESTED** [review]

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                temp = arr[j]
+                arr[j] = arr[j+1]
+                arr[j+1] = temp
+
+
+# initially list contains
+arr = [5, 4, 3, 2, 1]
+
+# sorting list
+bubbleSort(arr)
```
[the remark this answers is not in the document]
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:176]: okay done

[the remark this answers is not in the document]
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:176]: okay done

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+- Time Complexity:
+
+  - **Best Case**: O (n)
+  - **Average Case**: O (n<sup>2</sup>)
+  - **Worst Case**: O (n<sup>2</sup>)
+
+- Space Complexity: O (1)
+
+## Credits
```
[the remark this answers is not in the document]
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:209]: @nidh-ish I didn't copy code from NeoAlgo so can I give credit to myself or I remove this credit section?

File: docs/sorting-algorithms/bubble-sort.mdx (hunk 1 above)
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:14]: okay

File: docs/sorting-algorithms/bubble-sort.mdx (hunk 2 above)
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:92]: done

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+## Sample Input and Output
+
+### Input:
+
+```
+5
+5 4 3 2 1
+```
+
+### Output:
```
[the remark this answers is not in the document]
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:193]: done

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+print(arr)
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+### Input:
```
[the remark this answers is not in the document]
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:186]: done

File: docs/sorting-algorithms/bubble-sort.mdx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    element = int(input("Enter element : "))
+
+    arr.append(element)
+
+print("\nyou entered   :: ", arr)
+
+# sorting list
+bubbleSort(arr)
+
+print("After sorting :: ", arr)
```
Comment (nidh-ish) [inline docs/sorting-algorithms/bubble-sort.mdx:186]: These input statements are not present in C++ code. Make I/O similar across both codes.

**Review by nidh-ish — CHANGES_REQUESTED** [review]

File: docs/sorting-algorithms/bubble-sort.mdx (hunk 7 above)
Comment (ananddasani) [reply docs/sorting-algorithms/bubble-sort.mdx:186]: okay

**Comment by shonali2600** [comment]
@HarshCasper , @nidh-ish I too was assigned to the docs for bubble sort #206 and made the pull request #242 2 days ago. So please let me know that which PR will be considered, as I've also worked for the same and made the PR first. I'll accept any decision that will be made.

**Comment by ananddasani** [comment]
> @HarshCasper , @nidh-ish I too was assigned to the docs for bubble sort #206 and made the pull request #242 2 days ago. So please let me know that which PR will be considered, as I've also worked for the same and made the PR first. I'll accept any decision that will be made.

@shonali2600 As our rule of FCFS I was assigned this issue first on 8th April, so I think you can try to write a blog if any... 🤔

**Comment by shonali2600** [comment]
@ananddasani I totally agree, FCFS follows on getting assigned on the same issue, but we both were assigned to the different but same work, and I was not knowing that this issue has been created previously or someone has already been assigned to it. But I read somewhere in gssoc general that if more than one people working on the same issue then the person who made the PR first will be considered, and also taking 8 days for any issues is too long I guess. 

In some of the projects deadline is also given to the participant, and if any participant is not able to solve in that time period it gets assigned to other willing participant.

**Comment by HarshCasper** [comment]
Hi @shonali2600 @ananddasani 

I guess I will go with the PR created by @shonali2600. I'm sorry for this experience and the assignment, which is totally my fault. I hope you can pick up some other existing issue.

**Pull request closed without merging**